src/Files.App.Controls/ThemedIcon/ThemedIconLayer/ThemedIconLayer.Properties.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Microsoft.UI.Xaml;
+
+namespace Files.App.Controls
+{
+    public partial class ThemedIconLayer
+    {
+        //
+        // Dependency Property Registration
```

Comment:
The header is there to help group the types of properties

Reply:
Actually, I have a plan to use C# attributes.

```c#
// ThemedIcon.Properties.cs
[DependencyProperty(Name = "IconType", Type = typeof(ThemedIconType), Fallback = OnIconTypeChanged)]
public partial class ThemedIcon
{
}
``` 

That complex code turns into single code!

Reply:
But if want to keep it's ok since the control implementation won't be changed so frequently

Reply:
Source Generator catches that attributes and parses it.
- https://github.com/files-community/Files/blob/main/src/Files.Core.SourceGenerator/DependencyPropertyGenerator.cs
- https://github.com/files-community/Files/blob/main/src/Files.App/Data/Attributes/DependencyPropertyAttribute.cs
- https://github.com/files-community/Files/blob/main/src/Files.App/UserControls/PathBreadcrumb.xaml.cs#L10

Reply:
Are we able to move or copy those to the Controls project, so ThemedIcon and future controls can use them?

As long as functionality is not hampered, and it works well - it will simplify the code.